pallets/staking/src/lib.rs

```diff
@@ -873,12 +875,129 @@ pub mod weight {
+    fn reap_stash(s: u32) -> Weight;
+    fn new_era(v: u32, n: u32) -> Weight;
+    fn do_slash(l: u32) -> Weight;
+    fn payout_all(v: u32, n: u32) -> Weight;
+    fn submit_solution_initial(v: u32, n: u32, a: u32, w: u32) -> Weight;
+    fn submit_solution_better(v: u32, n: u32, a: u32, w: u32) -> Weight;
+    fn submit_solution_weaker(v: u32, n: u32) -> Weight;
+}
+
+impl WeightInfo for () {
+    fn bond(_u: u32) -> Weight {
+        1_000_000_000
+    }
+    fn bond_extra(_u: u32) -> Weight {
+        1_000_000_000
+    }
+    fn unbond(_u: u32) -> Weight {
+        1_000_000_000
+    }
+    fn withdraw_unbonded_update(_s: u32) -> Weight {
+        1_000_000_000
+    }
+    fn withdraw_unbonded_kill(_s: u32) -> Weight {
+        1_000_000_000
+    }
+    fn validate(_u: u32) -> Weight {
+        1_000_000_000
+    }
+    fn nominate(_n: u32) -> Weight {
+        1_000_000_000
+    }
+    fn chill(_u: u32) -> Weight {
+        1_000_000_000
+    }
+    fn set_payee(_u: u32) -> Weight {
+        1_000_000_000
+    }
+    fn set_controller(_u: u32) -> Weight {
+        1_000_000_000
+    }
+    fn set_validator_count(_c: u32) -> Weight {
+        1_000_000_000
+    }
+    fn force_no_eras(_i: u32) -> Weight {
+        1_000_000_000
+    }
+    fn force_new_era(_i: u32) -> Weight {
+        1_000_000_000
+    }
+    fn force_new_era_always(_i: u32) -> Weight {
+        1_000_000_000
+    }
+    fn set_invulnerables(_v: u32) -> Weight {
+        1_000_000_000
+    }
+    fn force_unstake(_s: u32) -> Weight {
+        1_000_000_000
+    }
+    fn cancel_deferred_slash(_s: u32) -> Weight {
+        1_000_000_000
+    }
+    fn payout_stakers(_n: u32) -> Weight {
+        1_000_000_000
+    }
+    fn payout_stakers_alive_controller(_n: u32) -> Weight {
+        1_000_000_000
+    }
+    fn rebond(_l: u32) -> Weight {
+        1_000_000_000
+    }
+    fn set_history_depth(_e: u32) -> Weight {
+        1_000_000_000
+    }
+    fn reap_stash(_s: u32) -> Weight {
+        1_000_000_000
+    }
+    fn new_era(_v: u32, _n: u32) -> Weight {
+        1_000_000_000
+    }
+    fn do_slash(_l: u32) -> Weight {
+        1_000_000_000
+    }
+    fn payout_all(_v: u32, _n: u32) -> Weight {
+        1_000_000_000
+    }
+    fn submit_solution_initial(_v: u32, _n: u32, _a: u32, _w: u32) -> Weight {
+        1_000_000_000
+    }
+    fn submit_solution_better(_v: u32, _n: u32, _a: u32, _w: u32) -> Weight {
+        1_000_000_000
+    }
+    fn submit_solution_weaker(_v: u32, _n: u32) -> Weight {
+        1_000_000_000
+    }
+}
+
 pub trait Trait:
     frame_system::Trait + SendTransactionTypes<Call<Self>> + pallet_babe::Trait + IdentityTrait
 {
     /// The staking balance.
-    type Currency: LockableCurrency<Self::AccountId, Moment = Self::BlockNumber>
-        + BlockRewardsReserveCurrency<BalanceOf<Self>, NegativeImbalanceOf<Self>>;
+    type Currency: LockableCurrency<Self::AccountId, Moment = Self::BlockNumber>;
```

Comment:
aren't we using the block reward reserve anymore? I thought we are still issuing rewards from reserve and if it is empty then we mint new tokens. 
cc @adamdossa @maxsam4

Reply:
If things compile without the extra bound then it must be unused.

Reply:
Yeah that makes sense but my question is more about the business decision, I don't remember about a task that removes this either it gets missed in other changes(upgrade or in any other tasks) or I am not aware of this change but this is highly unlikely as we still posses `BlockRewardsReserveCurrency::issue_using_block_rewards_reserve()`

Reply:
Ah, I see; let's bring it up on the grooming/standup. :)